io/shared/src/main/scala/scalaz/effect/IO.scala

```diff
@@ -728,27 +728,22 @@ object IO {
    * Submerges the error case of a disjunction into the `IO`. The inverse
    * operation of `IO.attempt`.
    */
-  final def absolve[E, A](v: IO[E, E \/ A]): IO[E, A] =
-    v.flatMap {
-      case -\/(e) => IO.fail(e)
-      case \/-(a) => IO.now(a)
-    }
+  final def absolve[E, A](v: IO[E, Either[E, A]]): IO[E, A] =
+    v.flatMap(_.fold[IO[E, A]](IO.fail, IO.now))
 
   /**
    * Retrieves the supervisor associated with the fiber running the action
    * returned by this method.
    */
-  def supervisor[E]: IO[E, Throwable => IO[Void, Unit]] = new Supervisor()
+  def supervisor[E]: IO[E, Throwable => IO[Nothing, Unit]] = new Supervisor()
 
   /**
    * Requires that the given `IO[E, Maybe[A]]` contain a value. If there is no
```

Comment:
We should check all the comments and docs and make sure they refer to the right types.